pyannote/audio/interactive/v03.py

```diff
@@ -0,0 +1,193 @@
+        Duration of chunks, in seconds. Defaults to 10s.
+    Yields
+    ------
+    task : dict
+        Prodigy task with the following keys:
+        "path" : path to audio file
+        "text" : name of audio file
+        "chunk" : chunk start and end times
+        "audio" : base64 encoding of audio chunk
+        "audio_spans" : speech spans detected by pretrained SAD model
+        "audio_spans_original" : copy of "audio_spans"
+        "meta" : additional meta-data displayed in Prodigy UI
+        "recipe" : "pyannote.sad.manual"
+    """
+    raw_audio = AudioFile(sample_rate=SAMPLE_RATE, mono=True)
+    # print(pipeline.onset)
+
+    for audio_source in Audio(source):
+
+        path = audio_source["path"]
+        text = audio_source["text"]
+        file = {"uri": text, "database": source, "audio": path}
+
+        duration = raw_audio.get_duration(file)
+        file["duration"] = duration
+
+        prodigy.log(f"RECIPE: detecting speech regions in '{path}'")
+
+        speech: Annotation = pipeline(file)
+
+        if duration <= chunk:
+            waveform, sr = raw_audio.crop(file, Segment(0, duration))
+            waveform = waveform.numpy()
+            task_audio = to_base64(normalize(waveform), sample_rate=SAMPLE_RATE)
+            task_audio_spans = to_audio_spans(speech)
+
+            yield {
+                "path": path,
+                "text": text,
+                "audio": task_audio,
+                "audio_spans": task_audio_spans,
+                "audio_spans_original": deepcopy(task_audio_spans),
+                "chunk": {"start": 0, "end": duration},
+                "meta": {"file": text},
+                # this is needed by other recipes
+                "recipe": "pyannote.sad.manual",
+            }
+
+        else:
+            for focus in chunks(duration, chunk=chunk, shuffle=False):
+                task_text = f"{text} [{focus.start:.1f}, {focus.end:.1f}]"
+                waveform, sr = raw_audio.crop(file, focus)
+                waveform = waveform.numpy().T
+                task_audio = to_base64(normalize(waveform), sample_rate=SAMPLE_RATE)
+                task_audio_spans = to_audio_spans(
+                    speech.crop(focus, mode="intersection"), focus=focus
+                )
+
+                yield {
+                    "path": path,
+                    "text": task_text,
+                    "audio": task_audio,
+                    "audio_spans": task_audio_spans,
+                    "audio_spans_original": deepcopy(task_audio_spans),
+                    "chunk": {"start": focus.start, "end": focus.end},
+                    "meta": {
+                        "file": text,
+                        "start": f"{focus.start:.1f}",
+                        "end": f"{focus.end:.1f}",
+                    },
+                    # this is needed by other recipes
+                    "recipe": "pyannote.sad.manual",
+                }
+
+
+@recipe(
+    "audio.v03",
```

Comment:
Let's call this recipe `pyannote.voice_activity_detection` or simply `voice_activity_detection`